### PR TITLE
Support linuxbrew

### PR DESCRIPTION
**Pull request by francoisferrand**
Allow installing DRY on linux using linuxbrew.

### CODE REVIEW

**Comment by moncho** [comment]
Thanks a lot, @Typz!!

**Comment by shoichiaizawa** [comment]
@Typz 

### TL;DR

Thanks for adding the Linuxbrew support. But this seems to break the macOS support. Only the Linuxbrew support works successfully with your latest commit (at least I can say this for Ubuntu Desktop 16.04, which I tested).

### Details

This morning (UTC+9), it was about 5 hours later @moncho merged this PR, I performed `brew update` and I saw the following error message:

```sh
$ brew update && brew upgrade && brew cleanup && brew doctor;
Already up-to-date.
Error: uninitialized constant OS::NAME
Please report this bug:
  https://docs.brew.sh/Troubleshooting
/usr/local/Homebrew/Library/Taps/moncho/homebrew-dry/dry.rb:7:in `<class:Dry>'
/usr/local/Homebrew/Library/Taps/moncho/homebrew-dry/dry.rb:3:in `load_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:26:in `module_eval'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:26:in `load_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:46:in `load_formula_from_path'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:101:in `load_file'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:239:in `load_file'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:92:in `klass'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:88:in `get_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:229:in `get_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:281:in `factory'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:319:in `from_keg'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:295:in `from_rack'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1407:in `block in installed'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1405:in `each'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1405:in `flat_map'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1405:in `installed'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:36:in `upgrade'
/usr/local/Homebrew/Library/Homebrew/brew.rb:99:in `<main>'
```

Initially, I had no clue what this above had been caused by; googling with the keyword `"Error: uninitialized constant OS::NAME"` didn't help me at all. But taking a close look at the error message, I realised that something had possibly gone wrong with the `dry.rb` formula.

So I ran `brew untap moncho/dry` to see if the error message would disappear – it did disappear. And then, I tried tapping the formula back by running `brew tap moncho/dry` to see if the tap would be fixed, but got the similar error message as the one above; please see the following error message:

```sh
$ brew tap moncho/dry
==> Tapping moncho/dry
Cloning into '/usr/local/Homebrew/Library/Taps/moncho/homebrew-dry'...
remote: Counting objects: 5, done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 5 (delta 0), reused 2 (delta 0), pack-reused 0
Unpacking objects: 100% (5/5), done.
Error: Invalid formula: /usr/local/Homebrew/Library/Taps/moncho/homebrew-dry/dry.rb
uninitialized constant OS::NAME
Error: Cannot tap moncho/dry: invalid syntax in tap!
```

Assuming that this above suggests that the OS detection script you added is missing something to retain the macOS support. Adding an extra note, the `dry` formula with your latest commit is tappable and installable on Ubuntu Desktop 16.04 (64 bit); there is no such error message output when tapping the formula.

---

All I can say is to inform the possible bug. I suppose this should easily be fixed if you know the Ruby language enough – unfortunatelly I don't 😢 

Would you be able to look into it? Including the fact that the error I mentioned above is actually reproducible or not.

Thanks for reading 😂

sho

**Comment by moncho** [comment]
Thanks for the detailed explanation, @shoichiaizawa!!

The Formula have been [changed](https://github.com/moncho/homebrew-dry/commit/0e606c39933f7fbd1864d176ca547bae0f9c2301) and it seems to be working fine on macOS.

@Typz could you please try the Formula again with Linuxbrew to check that I didnt broke anything?

**Comment by francoisferrand** [comment]
@moncho Yes, still working fine. Sorry about the regression.

**Comment by moncho** [comment]
@Typz thanks and no worries!
